launcher/src/main/java/org/apache/spark/launcher/CommandBuilderUtils.java

```diff
@@ -124,6 +124,9 @@ static JavaVendor getJavaVendor() {
     if (vendorString.contains("OpenJDK")) {
       return JavaVendor.OpenJDK;
     }
+    if (vendorString.contains("OpenJ9")) {
```

Comment:
Do you know who uses this `getJavaVendor` method?

As it is package private so only classes from the same package could access this.
Could it be a 3rd party tool should use the same package name: `org.apache.spark.launcher`?

As in the entire Spark code there is no usage for it:
```
$ find . -type f -exec grep "getJavaVendor" \{} \; -print
  static JavaVendor getJavaVendor() {
./launcher/src/main/java/org/apache/spark/launcher/CommandBuilderUtils.java
```

Reply:
Hm yeah it's unused as far as I can see. It shouldn't be used by third parties anyway.

Reply:
There is one more method near this one also seams to me unused: `firstNonEmptyValue`. Should I create a Minor PR or OK to remove within this PR?

Reply:
Oh and there will be the enum as well: `getJavaVendor` seams to me the only place where used (and mima exclude).

Reply:
Agree, you can delete the unused code here.
It looks like this was added to account for the fact that IBM JDKs don't have a PermGen, but Spark no longer tries to set this now that Java 8 is required and doesn't have a PermGen.